src/components/Flyout/FlyoutWindow.part.tsx

```diff
@@ -354,6 +354,8 @@ export class FlyoutWindowInt extends React.Component<FlyoutWindowProps, FlyoutWi
   private setFlyoutRef = (el: HTMLDivElement) => {
     if (el) {
       this.flyoutContainer = el;
+      this.flyoutContainer.removeEventListener('scroll', this.onScroll);
```

Comment:
This wont work (if we really changed the node you are using the new el - which hasnt any listener appended yet).

Hopefully the following code fragment shows you the essential idea I tried to explain:

```jsx
//...

private setRef = (el: Element) => {
  if (this.container) {
    this.container.removeEventListener(...);
  }

  if (el) {
    el.addEventListener(...);
  }

  this.container = el;
}
```

(no cleanup in `componentWillUnmount` needed in this case)